content_scripts/mode_find.coffee

```diff
@@ -0,0 +1,44 @@
+# NOTE(smblott).  Ultimately, all of the FindMode-related code should be moved to this file.
+
+# When we use find mode, the selection/focus can end up in a focusable/editable element.  In this situation,
+# PostFindMode handles two special cases:
+#   1. Be an InsertModeBlocker.  This prevents keyboard events from dropping us unintentionaly into insert
+#   mode.  Here, this is achieved by inheriting from InsertModeBlocker.
+#   2. If the very-next keystroke is Escape, then drop immediately into insert mode.
+#
+class PostFindMode extends InsertModeBlocker
```

Comment:
@philc The only way to implement #556 without it clearing the selection when we focus the input is to have it in an iframe. I've done this in #1401 (although it's actually a `contentEditable` element there, so it resizes with its contents).